src/fidesops/schemas/shared_mixins.py

```diff
@@ -0,0 +1,30 @@
+from typing import Dict, Any, List
+
+from fideslang.validation import FidesKey, FidesValidationError
+from pydantic import root_validator
+from pydantic.main import BaseModel
+
+
+class FidesKeyMixin(BaseModel):
+    """
+    Abstract mixin to validate fields of type FidesKey
+    Converts FidesValidationError to api-friendly ValueError
+    """
+
+    _fides_key_field_names: List[str] = ["key"]
+
+    @root_validator()
```

Comment:
Try adding `pre=True` here -

The current solution doesn't prevent unhandled 500 exceptions because `FidesKey` itself has validation on it. Because individual fields are validated before the `root_validator` runs, you will still get 500 exceptions that were raised before this code was even hit. I believe `pre=True` lets the root validator run first.

Reply:
Also, could you add some unit tests around invalid keys used on some of the relevant API endpoints?  Doesn't look like we have any test coverage of this specifically.